src/Tools/dotnet-monitor/DiagnosticPortPostConfigureOptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal sealed class DiagnosticPortPostConfigureOptions :
+        IPostConfigureOptions<DiagnosticPortOptions>
+    {
+        private readonly IConfiguration  _configuration;
+
+        public DiagnosticPortPostConfigureOptions(
+            IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void PostConfigure(string name, DiagnosticPortOptions options)
```

Comment:
You'll want to write a unit test to validate this once #1372 is merged: setup configuration, bind DiagnosticPortOptions, validate values for difference combinations that exercise this code.

Reply:
Sounds good; do you want me to hold this PR for that, or check this in and then do a separate PR for the test code?

Reply:
You can do a separate PR for it, but I would say get it done soon so we don't add technical debt long term.